packages/backend/src/database/migrations/20241129140659_add-table-name-to-catalog-search.ts

```diff
@@ -0,0 +1,35 @@
+import { Knex } from 'knex';
+
+const CATALOG_SEARCH_TABLE = 'catalog_search';
+const CACHED_EXPLORE_TABLE = 'cached_explore';
+
+export async function up(knex: Knex): Promise<void> {
+    // setting to nullable to allow for migration of existing data
+    await knex.schema.alterTable(CATALOG_SEARCH_TABLE, (table) => {
+        table.text('table_name').nullable();
+    });
+
+    // backfill table_name with data from cached_explore
+    await knex(CATALOG_SEARCH_TABLE).update({
+        // @ts-ignore - we're doing something similar in 20240604094543_add_views_columns_to_chart_and_dashboards_tables.ts
+        table_name: knex
+            .select(knex.raw(`explore->>'baseTable'`))
+            .from(CACHED_EXPLORE_TABLE)
+            .whereRaw(
+                `${CACHED_EXPLORE_TABLE}.cached_explore_uuid = ${CATALOG_SEARCH_TABLE}.cached_explore_uuid`,
+            )
+            .limit(1),
+    });
```

Comment:
will this be a problematic because of slow migration for app cloud?

Reply:
It should be possible to do this update with a single query and not `N` subqueries, that would make it significantly faster! I'm not sure it's supported with knex syntax but with knex raw you would do:

```sql
UPDATE catalog_search
SET table_name = cached_explore.explore->>baseTable
FROM cached_explore
WHERE catalog_search.cached_explore_uuid = cached_explore.cached_explore_uuid
```

This assumes:
1. That `catalog_search` is `m:1` with `cached_explore`
2. And that `cached_explore` has all the entries for `catalog_search` maybe that's not true?

Reply:
Amazing! I have updated the code using `knex.raw` as `updateFrom` PostgreSQL specific `updateFrom` only got introduced in knex 2.5.0 https://github.com/knex/knex/releases/tag/2.5.0